pallets/automation-time/src/tests.rs

```diff
@@ -2776,32 +3133,62 @@ fn on_init_runs_tasks() {
 			vec![40],
 			vec![SCHEDULED_TIME],
 			Action::Notify { message: message_one.clone() },
+			vec![],
 		);
 		let message_two: Vec<u8> = vec![2, 4];
 		let task_id2 = add_task_to_task_queue(
 			ALICE,
 			vec![50],
 			vec![SCHEDULED_TIME],
 			Action::Notify { message: message_two.clone() },
+			vec![],
 		);
 		let task_id3 = add_task_to_task_queue(
 			ALICE,
 			vec![60],
 			vec![SCHEDULED_TIME],
 			Action::Notify { message: vec![50] },
+			vec![],
 		);
 		LastTimeSlot::<Test>::put((LAST_BLOCK_TIME, LAST_BLOCK_TIME));
 
 		AutomationTime::on_initialize(1);
+
+		let mut condition: BTreeMap<String, String> = BTreeMap::new();
+		condition.insert(String::from("type"), String::from("time"));
+		condition.insert(String::from("timestamp"), format!("{}", LAST_BLOCK_TIME));
+
 		assert_eq!(
 			events(),
 			[
+				RuntimeEvent::AutomationTime(crate::Event::TaskTriggered {
+					who: owner.clone(),
+					task_id: task_id1.clone(),
+					condition: condition.clone(),
+					encoded_call: vec![],
+					abort_errors: vec![],
+				}),
 				RuntimeEvent::AutomationTime(crate::Event::Notify { message: message_one.clone() }),
+				RuntimeEvent::AutomationTime(crate::Event::TaskExecuted {
+					who: owner.clone(),
+					task_id: task_id1.clone(),
+				}),
 				RuntimeEvent::AutomationTime(crate::Event::TaskCompleted {
```

Comment:
`Event::TaskCompleted` shouldn’t fire here. It fires only at the very end.

Reply:
The same applies to the below tasks.

Reply:
Yes, It indeed triggers when the task is completed.

There are 3 tasks in this test, and each task is executed only once. So, after each task is executed, the Event::TaskCompleted will be triggered.

Reply:
okay, please add comments here to indicated that the first task is completed and the subsequent events are for the rest of tasks.

Reply:
Fixed